packages/block-editor/src/hooks/style.js

```diff
@@ -181,6 +202,19 @@ const skipSerializationPathsSave = {
 	[ `${ SPACING_SUPPORT_KEY }` ]: [ 'spacing.blockGap' ],
 };
 
+/**
+ * A dictionary used to normalize feature names between support flags, style
+ * object properties and __experimentSkipSerialization configuration arrays.
+ *
+ * This allows not having to provide a migration for a support flag and possible
+ * backwards compatibility bridges, while still achieving consistency between
+ * the support flag and the skip serialization array.
+ *
+ * @constant
+ * @type {Record<string, string>}
+ */
+const renamedFeatures = { gradients: 'gradient' };
```

Comment:
All but `color.gradients` map neatly between the support flag and the property within the style object. In the case of `gradients` the property that needs to be omitted from the styles is `gradient`. 

I ended up going with the option here of explicitly mapping this. We can therefore keep the flag to opt in to support the same as the value needed to skip serialization of just the `gradient` feature without having to migrate the support flag or style object property etc.

If there are better options, I'll happily adopt them.

Reply:
I think this is nice way forward for the sake of an outlying plural 😄 and to keep the block.json keys predictable and consistent.

Having the same key for `color.gradients` to `color.__experimentalSkipSerialization.gradients`, in my opinion, is worth it:

```json
		"color": {
			"gradients": true,
			"__experimentalSkipSerialization": [ "gradients" ],
                        ...
```